src/SprykerEco/Zed/Adyen/Business/Hook/Saver/MakePayment/AbstractSaver.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Adyen\Business\Hook\Saver\MakePayment;
+
+use Generated\Shared\Transfer\AdyenApiRequestTransfer;
+use Generated\Shared\Transfer\AdyenApiResponseTransfer;
+use SprykerEco\Zed\Adyen\AdyenConfig;
+use SprykerEco\Zed\Adyen\Business\Reader\AdyenReaderInterface;
+use SprykerEco\Zed\Adyen\Business\Writer\AdyenWriterInterface;
+
+abstract class AbstractSaver
+{
+    /**
+     * @var \SprykerEco\Zed\Adyen\Business\Reader\AdyenReaderInterface
+     */
+    protected $reader;
+
+    /**
+     * @var \SprykerEco\Zed\Adyen\Business\Writer\AdyenWriterInterface
+     */
+    protected $writer;
+
+    /**
+     * @var \SprykerEco\Zed\Adyen\AdyenConfig
+     */
+    protected $config;
+
+    /**
+     * @return string
+     */
+    abstract protected function getRequestType(): string;
```

Comment:
maybe move this below the constructor?

Reply:
I guess we have a rule that abstract methods have to be in top.